src/solver/conv_asm_implicit_gemm_bwd_v4r1_dynamic.cpp

```diff
@@ -0,0 +1,202 @@
+           (dilation_w == 1) && (pad_h == 0) && (pad_w == 0))
+        {
+            if((gemm_m % 128 == 0) && (gemm_n % 128 == 0) && (k % 8 == 0) && ((ho * wo) % 16 == 0))
+            {
+                grid_size   = (gemm_m >> 7) * (gemm_n >> 7);
+                block_size  = 256;
+                kernel_name = "igemm_bwd_gtc_bt128x128x8_tt8x8_gm2x4x4_gn2x4x4_ta1x1x1x1x4_"
+                              "8x1x1x32x1_tb1x1x1x1x4x1x1_8x1x1x2x1x1x16";
+                return true;
+            }
+            else if((gemm_m % 64 == 0) && (gemm_n % 64 == 0) && (k % 8 == 0) && (n % 64 == 0))
+            {
+                grid_size   = (gemm_m >> 6) * (gemm_n >> 6);
+                block_size  = 64;
+                kernel_name = "igemm_bwd_gtc_bt64x64x8_tt8x8_gm2x4x2_gn2x4x2_ta1x2x1x1x4_"
+                              "4x1x1x16x1_tb1x2x1x1x4x1x1_4x1x1x16x1x1x1";
+                return true;
+            }
+        }
+        else
+        {
+            // TODO: add here!
+        }
+    }
+    return false;
+}
+
+bool ConvAsmImplicitGemmV4R1DynamicBwd::IsApplicable(const ConvolutionContext& ctx) const
+{
+    const auto device_name = ctx.GetStream().GetDeviceName();
+    if(!(StartsWith(device_name, "gfx900") || StartsWith(device_name, "gfx906")))
+        return false;
+
+    if(!ctx.direction.IsBackwardData())
+        return false;
+
+    if(!ctx.Is2d())
+        return false;
+
+    if(!ctx.IsFp32())
+        return false;
+
+    if(!ctx.rmv.IsV3())
+        return false;
+
+    if(ctx.group_counts != 1)
+        return false;
+
+    std::string kernel_name;
+    int block_size;
+    int grid_size;
+    return FindImplicitGemmDynamicKernelBwd(ctx, kernel_name, block_size, grid_size);
+}
+
+ConvSolution ConvAsmImplicitGemmV4R1DynamicBwd::GetSolution(const ConvolutionContext& ctx) const
+{
+    ConvSolution result;
+
+    std::string kernel_name;
+    int block_size;
+    int grid_size;
+    bool ret = FindImplicitGemmDynamicKernelBwd(ctx, kernel_name, block_size, grid_size);
+    if(!ret)
+        MIOPEN_THROW("should not happen!");
+
+    KernelInfo kernel;
+    std::ostringstream options;
+
+    kernel.kernel_file = "igemm_bwd_gtc_dynamic.s";
+    kernel.kernel_name = kernel_name;
+    kernel.g_wk.clear();
+    /* Note here, for API like hipHccModuleLaunchKernel(), hipExtModuleLaunchKernel()
+    * grid dims is in unit of work item.
+    * But for api like hipModuleLaunchKernel(), grid dim is in unit of block.
+    */
+    kernel.g_wk.push_back(grid_size * block_size);
+    kernel.g_wk.push_back(1);
+    kernel.g_wk.push_back(1);
+    kernel.l_wk.clear();
+    kernel.l_wk.push_back(block_size);
+    kernel.l_wk.push_back(1);
+    kernel.l_wk.push_back(1);
+
+    GenerateClangDefsym(options, "ROCM_METADATA_VERSION", ctx.rmv.UseV3() ? 5 : 4);
+
+    kernel.comp_options = options.str();
+
+    MIOPEN_LOG_I2(kernel.kernel_file + ":" + kernel.kernel_name);
```

Comment:
Remove this log as when you increase log level to 6/7, you will get to see this info through other logs. You won't need it

Reply:
Sure